processes/addFulfillManifest.py

```diff
@@ -0,0 +1,140 @@
+import json
+import os
+import boto3
+import copy
+import logging
+from requests.exceptions import HTTPError, ConnectionError
+from botocore.exceptions import ClientError
+
+from .core import CoreProcess
+from urllib.error import HTTPError
+from managers import WebpubManifest
+from datetime import datetime, timedelta, timezone
+from managers import DBManager
+from model import Link
+from logger import createLog
+
+logger = createLog(__name__)
+
+s3_client = boto3.client("s3")
+bucketName = 'drb-files-qa'
+
+class FulfillProcess(CoreProcess):
+
+    def __init__(self, *args):
+        super(FulfillProcess, self).__init__(*args[:4])
+
+        self.fullImport = self.process == 'complete' 
+        self.startTimestamp = None
+
+        # Connect to database
+        self.generateEngine()
+        self.createSession()
+
+        # S3 Configuration
+        self.s3Bucket = os.environ['FILE_BUCKET']
+        self.createS3Client()
+
+    def runProcess(self):
+        if self.process == 'daily':
+            startTimeStamp = datetime.now(timezone.utc) - timedelta(days=1)
+            self.updateManifest(startTimeStamp)
+        elif self.process == 'complete':
+            self.updateManifest()
+        elif self.process == 'custom':
+            timeStamp = self.ingestPeriod
+            startTimeStamp = datetime.strptime(timeStamp, '%Y-%m-%dT%H:%M:%S')
+            self.updateManifest(startTimeStamp)
+
+    def updateManifest(self, startTimeStamp=None):
+
+        '''Load batch of LA works from past 24 hours'''
+
+        batches = self.load_batch()
+        for batch in batches:
+            for c in batch['Contents']:
+                if c["LastModified"] > startTimeStamp.replace(tzinfo=timezone.utc):
+                    currKey = c['Key']
+                    metadataObject = self.getObjectFromBucket(currKey, self.s3Bucket)
+                    self.update_batch(metadataObject, self.s3Bucket, currKey)
+
+    def update_batch(self, metadataObject, bucketName, currKey):
```

Comment:
I'd rename this function since it only deals with one manifest at a time (this implies it deals with the whole batch)